Revert PR #17895

rodrigoprimo opened the pull request:
### Changes proposed in this Pull Request:

This PR reverts #17895 which was introduced in WC 3.5 to use post_author instead of `_customer_user` post meta to store customer ID for orders. After talking to @claudiulodro, we decided to revert the changes for WC 3.5.1 and reconsider #17895 for 3.6. This decision was made due to the bugs that were introduced by #17895 and that we didn't catch before the release (#21656, #21687 and #21737). All the issues seem fixable, but we are opting to err on the side of caution instead of rushing to create fixes that could introduce more problems. Reverting is the safest and fastest option that we have now.

#17895 copied data from `_customer_user` postmeta to the field `post_autor` of the table `wp_posts` and changed WooCommerce to populate both `customer_user` and `post_author` when a order is created or updated. So, this PR simply revert the code changes to make WooCommerce get the customer ID for orders in the old way. For now, we are not touching the data that was copied to `post_author` and this shouldn't be a problem as WooCommerce doesn't use this field for orders.

It was not possible to revert #17895 automatically, so I had to revert each commit manually. I decided to keep the changes introduced in `includes/class-wc-background-updater.php` (https://github.com/woocommerce/woocommerce/pull/17895/files#diff-3809cc9b6218b1f645071ed0351b2edc) as they could be useful for future database update routines.

Closes #21656, #21687, #21737 and #21741.
Supersedes #21673 and #21695

### How to test the changes in this Pull Request:

1. Check that code changes were properly reverted
2. Check that the issue in #21737 was fixed
3. Test that the functionality altered in this PR still works: customer reports, filter orders by customer in the admin interface, filter orders by customer in the REST API, saving customer ID when placing an order or manually created it via wp-admin, saving customer ID when updating an order, get customer last order (https://github.com/woocommerce/woocommerce/compare/revert/17895?expand=1#diff-d6dba1e5e24baa6d5d15671305dd1994R325), `wc_reset_order_customer_id_on_deleted_user()` and `wc_customer_bought_product()`.
4. Make sure that orders that were created after the WC 3.5 have the correct customer after updating to 3.5.1.

### Changelog entry

> Revert PR to use post_author instead of the post meta '_customer_user' to store customer ID for orders

rodrigoprimo committed:
Revert "Add @return for wc_update_350_order_customer_id() docblock"

This reverts commit 5c5cddf26ae1f84490fecef5ff828c3d13367c65.

rodrigoprimo committed:
Revert "Fix indentation"

This reverts commit 397e40f1659e38627be916c6f0114b33d0c2cf2c.

rodrigoprimo committed:
Revert "Pass user_id via $wpdb->prepare()"

This reverts commit 77add4670573b79634c4b192a01932a1a7569eb6.

rodrigoprimo committed:
Revert "Fix some code untitentional code changes added during the last merge from master"

This reverts commit 7f70c6372c32395309d28e5ad9afbda5ccd8cf2b.

rodrigoprimo committed:
Revert "Update WC_Order_Data_Store_CPT::get_wp_query_args() to use post_author instead of _customer_user"

This reverts commit 8db3236889c4edbc34df57574958b55704ac03e7.

rodrigoprimo committed:
Revert "Update WC_Customer_Data_Store to use post_author instead of _customer_user"

This reverts commit 482e844f16e396f30558d74045328c9bbb5f2e4d.

rodrigoprimo committed:
Revert "Update WC API v1 and v2 to use post_author instead of _customer_user"

This reverts commit fdfe95b07e2756118d6d7ffd955e00fd08550db2.

rodrigoprimo committed:
Revert "Update wc_customer_bought_product() and customers report to use post_author instead of _customer_user"

This reverts commit b97f2749d6851f7abbca737eb4a01400826670f7.

rodrigoprimo committed:
Revert "Set post_author to 0 on all orders that belong to a deleted customer"

This reverts commit f725dc17aff453f4e21b086393d7a4b154c40530.

rodrigoprimo committed:
Revert "Cache query to get the list of orders placed by the user ID 1"

This reverts commit 697421a9fae11c317230b3762224bc43947a294c.

rodrigoprimo committed:
Revert "Monitor only memory limit while running the data migration"

This reverts commit 923eeb2e43130575899823f078e6d0d86b3afe2f.

rodrigoprimo committed:
Revert "Migrate data to post_author in a single go when using WP-CLI"

This reverts commit e02f3fbec47dbc00a8d4b657d06490ba36591993.

rodrigoprimo committed:
Revert "Punt to WC version 3.5.0"

This reverts commit ccf10f6ab279902f1ce748df63cee1f769835f26.

rodrigoprimo committed:
Revert "Migrate _customer_user to post_author in batches"

This reverts commit 4123cc6f7de74176a325e9f316784c25fb12eff9.

rodrigoprimo committed:
Revert "Reposition wc_update_340_order_customer_id() function"

This reverts commit 35998e56cbe3939cb7a2eae2ce90c9a2b137bcb1.

rodrigoprimo committed:
Revert "Add filter to list of post_types used to build the query to copy _customer_user to post_author"

This reverts commit 1f39f7d63378caa86002795f19d2dae8890dc971.

rodrigoprimo committed:
Revert "Bump WC version to 3.4.0"

This reverts commit 7f43c39f2fd1ee0c0e2dc47ac0badbd749416f8b.

rodrigoprimo committed:
Revert "Use post_author field instead of post meta in orders admin page"

This reverts commit 6ecc3bc44229cf8f11bc071fd1546379a11f6a5d.

rodrigoprimo committed:
Revert "Make sure post_author is used only after WC 3.3.0 DB upgrade routine is executed"

This reverts commit 997a6fa40dab53247659ae5919895e9bbe0f47b9.

rodrigoprimo committed:
Revert "Set post_author to 0 for refund orders"

This reverts commit ad4304450d0ef2eda621f562d67f6e9f39c2bb1a.

rodrigoprimo committed:
Revert "First pass at using post_author instead of the post meta '_customer_user'"

This reverts commit 61c7abce4f50146fa595f6429e8f8d6fc082d9fc.

rodrigoprimo committed:
Preserve changes in includes/class-wc-background-updater.php

claudiulodro commented on includes/data-stores/class-wc-order-data-store-cpt.php:
```diff
@@ -265,9 +258,10 @@ protected function update_post_meta( &$order ) {
-		if ( in_array( 'billing_email', $updated_props ) ) {
-			$this->update_downloadable_permissions( $order );
+		// If customer changed, update any downloadable permissions.
+		if ( in_array( 'customer_user', $updated_props ) || in_array( 'billing_email', $updated_props ) ) {
```
This should be [customer_id](https://github.com/woocommerce/woocommerce/pull/17895/files#diff-9b0af3108fffa9d52d09b6504882fd0aL262)

claudiulodro reviewed (changes requested):
Looks good and works well in my testing. Just one small change needed.

rodrigoprimo committed:
Reapply change made in c765ce644db14e322fc77b0d3863d2e4160ecd4f

Reverting commit 61c7abc in commit ece9f02776626e7bff03ee635b1ce6129e1daae3, overrode a unrelated change made to the same code in another PR by commit c765ce644db14e322fc77b0d3863d2e4160ecd4f. This commit simply reapplies this change.

rodrigoprimo commented:
@claudiulodro, fce22a7 adds the change you requested. When I reverted each of the commits in this PR individually, I didn't revert merge commits that were used to keep #17895 branch in sync with master. When commit 61c7abce4f50146fa595f6429e8f8d6fc082d9fc was made a few months ago, there was a bug in the code and it was using`customer_user` instead of `customer_id`. This bug was fixed in a unrelated commit (https://github.com/woocommerce/woocommerce/commit/c765ce644db14e322fc77b0d3863d2e4160ecd4f#diff-9b0af3108fffa9d52d09b6504882fd0aL262) and the fixed was overrode when I created ece9f02776626e7bff03ee635b1ce6129e1daae3.
